### PR TITLE
Pla 12511/rename predictor

### PR DESCRIPTION
**Pull request by bfolie**
# Citrine Python PR

## Description 
Adds the ability to rename a predictor.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [ ] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in __version\__.py

### CODE REVIEW

File: src/citrine/resources/predictor.py [hunk 1]
```diff
@@ -596,3 +610,18 @@ def retrain_stale(self, uid: Union[UUID, str], *, version: Union[int, str]) -> G
...
+    def rename(self,
+               uid: Union[UUID, str],
+               *,
+               version: Union[int, str],
+               name: str,
+               description: str) -> GraphPredictor:
+        """Rename an existing predictor.
+
+        Both the name and description can be changed. This does not trigger retraining.
+        Any existing version of the predictor can be renamed, or "most recent".
```
Comment (bfolie) [inline src/citrine/resources/predictor.py:623]:
```suggestion
        Any existing version of the predictor can be renamed, or "most_recent".
```

File: src/citrine/resources/predictor.py [hunk 2]
```diff
@@ -596,3 +610,18 @@ def retrain_stale(self, uid: Union[UUID, str], *, version: Union[int, str]) -> G
         result in an error.
         """
         return self._versions_collection.retrain_stale(uid, version=version)
+
+    def rename(self,
+               uid: Union[UUID, str],
+               *,
+               version: Union[int, str],
+               name: str,
+               description: str) -> GraphPredictor:
```
Comment (anoto-moniz) [inline src/citrine/resources/predictor.py:619]: This caused me to realize the `/rename` endpoint requires both fields, which seems clunky and unlikely to be what users want. So I wrote up a ticket to make them optional: https://citrine.atlassian.net/browse/PLA-12705

**Review by anoto-moniz — APPROVED** [review]
LGTM! And we can update the interface without breaking anyone once we make both fields optional.

File: src/citrine/resources/predictor.py (hunk 2 above)
Comment (bfolie) [reply src/citrine/resources/predictor.py:619]: I agree

**Pull request merged**